PiwikTracker/Logger.swift

```diff
@@ -0,0 +1,55 @@
+import Foundation
+
+public enum LogLevel: Int {
+    case verbose = 10
+    case debug = 20
+    case info = 30
+    case warning = 40
+    case error = 50
+    
+    var shortcut: String {
+        switch self {
+        case .error: return "E"
+        case .warning: return "W"
+        case .info: return "I"
+        case .debug: return "D"
+        case .verbose: return "V"
+        }
+    }
+}
+
+public protocol Logger {
+    var minLevel: LogLevel { get set }
+    func log(message: ()->(String), with level: LogLevel)
+}
+
+extension Logger {
+    func verbose(message: ()->(String)) {
+        log(message: message, with: .verbose)
+    }
+    func debug(message: ()->(String)) {
+        log(message: message, with: .debug)
+    }
+    func info(message: ()->(String)) {
+        log(message: message, with: .info)
+    }
+    func warning(message: ()->(String)) {
+        log(message: message, with: .warning)
+    }
+    func error(message: ()->(String)) {
+        log(message: message, with: .error)
+    }
+}
+
+final class DefaultLogger: Logger {
```

Comment:
I rather like to use structs for data structures whose main task is to hold data. For things like this, where the main task is to "do things" or "define behavior" I like to use classes. What do you think about this?